packages/data-stores/src/purchases/hooks/use-site-purchase-by-id.ts

```diff
@@ -0,0 +1,17 @@
+import useSitePurchases from '../queries/use-site-purchases';
+import type { Purchase } from '../types';
+
+interface Props {
+	purchaseId?: number | null;
+	siteId?: string | number | null;
+}
+
+const useSitePurchaseById = ( { siteId, purchaseId }: Props ): Purchase | undefined => {
+	const sitePurchases = useSitePurchases( { siteId } );
+
+	return sitePurchases?.data && typeof purchaseId === 'number'
+		? sitePurchases?.data[ purchaseId ]
+		: undefined;
+};
```

Comment:
and surprisingly, not using `select` for this. The alternative (using `select`), while keeping this a simple hook (i.e. not returning `useSitePurchases` with the granular selector) doesn't add much or make it any better.